File: root/branches/2.0.1/source/modules/strategy/src/test/sample_data/inputs/part2.rb
include_class "org.marketcetera.strategy.ruby.Strategy"
class Part2 < Strategy
  def on_start
      @helper = Helper.new
      @helper.do_something_common
      @helper.do_something_part1
      @helper.do_something_part2
  end
  def on_stop
      @helper.do_something_common
      @helper.do_something_part1
      @helper.do_something_part2
  end
end

class Helper
    def do_something_common
    end  
  def do_something_part2
  end  
end

File: 20121022/modules/strategy/src/test/sample_data/inputs/Part2.java
import org.marketcetera.strategy.java.Strategy;


/* $License$ */

/**
 * Test strategy (part two of two parts) that tests what happens when a strategy helper-class is redefined. 
 *
 * @version $Id: Part2.java 16063 2012-01-31 18:21:55Z colin $
 * @since 1.0.0
 */
public class Part2
        extends Strategy
{
    private Helper helper;
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStart()
     */
    @Override
    public void onStart()
    {
        helper = new Helper();
        helper.doSomethingCommon();
        helper.doSomethingPart2();
    }
    /* (non-Javadoc)
     * @see org.marketcetera.strategy.java.Strategy#onStop()
     */
    @Override
    public void onStop()
    {
        helper.doSomethingCommon();
        helper.doSomethingPart2();
    }

    private static class Helper
    {
        private void doSomethingCommon()
        {
        }  
        private void doSomethingPart2()
        {
        }
    }
}
